pkg/cluster/internal/providers/docker/network.go

```diff
@@ -42,13 +47,15 @@ const fixedNetworkName = "kind"
 
 // ensureNetwork checks if docker network by name exists, if not it creates it
 func ensureNetwork(name string) error {
-	// TODO: the network might already exist and not have ipv6 ... :|
-	// discussion: https://github.com/kubernetes-sigs/kind/pull/1508#discussion_r414594198
-	exists, err := checkIfNetworkExists(name)
+	// check if network exists already and remove any duplicate networks
+	exists, err := removeDuplicateNetworks(name)
```

Comment:
at this point is hard to guess what is the good one, if there are multiple networks, there should have been created previously.
I think that this case is complex to handle and maybe we should just check if a network exists and pick one,

Reply:
The TODO is not new, only moved up.

This TODO is actually about the case that we previously created a network and it may have suboptimal config (like missing the ipmasquerade flag we added).

In this case we will NOT run the risk of duplicate networks (this can only happen if they're created concurrently) but we currently will continue to use the old suboptimal config.

What would possibly be ideal is something like: when the network is ensured we diff the settings vs desired and then automatically recreate it if it doesn't match and nothing is currently on the network. It needs some thought and still isn't handled, so the TODO was preserved when hoisting the existence check into this network list call here.

Reply:
if anybody is running with suboptimal config and didn't notice is that is optimal for him :smile: 

I will keep going with the TODO and iterate with more feedback, unless you had feedback about this and/or saw this behavior causing issues

Reply:
> if anybody is running with suboptimal config and didn't notice is that is optimal for him 😄

I mean some users are not aware of kind managing the network. In the future it might be good to at least hint that they should have it recreated.